### DISCUSSION TITLE
Radio Group

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#222] [Component suggestions]
<img width="480" alt="visuele weergave van het radio group component" src="https://github.com/nl-design-system/backlog/assets/545207/545d87f6-503a-491c-96f6-eeb55f2d7943">

We willen graag meer leren over de Radio Group component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1708952359928209)

---

## Naam
Radio Group

## Doel
Biedt de mogelijkheid om één optie te selecteren uit een lijst.

## Gerelateerde componenten
[Radio Button](https://github.com/orgs/nl-design-system/discussions/288), [Fieldset](https://github.com/orgs/nl-design-system/discussions/309), [Checkbox Group](https://github.com/orgs/nl-design-system/discussions/223), [Select](https://github.com/orgs/nl-design-system/discussions/239)

## Onderzoek
[Radio Group issue - NL Design System](https://github.com/nl-design-system/backlog/issues/129)
Het issue op de backlog bevat al 2 comments.
- Beschrijving van keyboard gedrag door radio buttons door Jules Ernst.
- Oproep om tot richtlijn te komen voor pre-selecting van radio optie of juist niet door Robbert.

[Radio Buttons: Select One by Default or Leave All Unselected? - Nielsen Norman Group](https://www.nngroup.com/articles/radio-buttons-default-selection/)
Artikel over pre-selecting van een Radio optie.

[Checkboxes vs. Radio Buttons - Nielsen Norman Group](https://www.nngroup.com/articles/checkboxes-vs-radio-buttons/)
Artikel over verschil tussen Checkboxes en Radio Buttons.

[We've updated the radios and checkboxes - GOV.UK](https://designnotes.blog.gov.uk/2016/11/30/weve-updated-the-radios-and-checkboxes-on-gov-uk/)
Tim Paul legt uit waarom GOV.UK aanpassingen heeft gedaan bij Radio en Checkbox opties.

[Radios - GOV.UK](https://design-system.service.gov.uk/components/radios/)
Onder 'Research on this component' worden bevindingen genoteerd. Er is ook een [link naar de Radios Github discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/59) waar meer informatie, onderzoek en aanpassingen worden gedeeld.

In een [comment bij Radio Github Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/59#issuecomment-871512686) wordt meer tekst en uitleg gegeven wat GOV.UK heeft gedaan rondom het conditioneel onthullen van een text input in combinatie met een Radio Group.

[Myth #23: Choices should always be limited to 7+/-2 - UX Myths](https://uxmyths.com/post/931925744/myth-23-choices-should-always-be-limited-to-seven)
Artikel beschrijft de mythe rondom het aanhouden van maximaal 7 opties.

**Onderzoek: Vergunning aanvragen - Gemeente Den Haag**
Bovenstaande mythe werd ook ontkracht tijdens dit onderzoek. Bij een meerstappenformulier bestond 1 stap uit een Radio Group met 11 opties. Hierbij waren geen bijzonderheden. Respondenten vonden de vragen en informatie duidelijk en volledig.

<img width="520" alt="screenshot van formulier stap met een radio group van 11 opties." src="https://github.com/user-attachments/assets/f9edac74-c5d3-4247-9d40-74c520bc94ed" />

**Comment by sergei-maertens** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/story/pure-react-components-forms-radiofield--default)

---

### Screenshots / Varianten

**Storybook**
Default
![Scherm­afbeelding 2024-02-27 om 15 27 43](https://github.com/nl-design-system/backlog/assets/545207/4a4cc21e-864a-4de7-bc6b-6928123159df)

Validatie
![Scherm­afbeelding 2024-02-27 om 15 27 52](https://github.com/nl-design-system/backlog/assets/545207/4a00680f-5bf8-40a9-bfe0-66db5df0bfdf)


---

### Notities
- Ik merk nu dat de styling behoorlijk vernaggeld is :thinking:

**Comment by Rozerinay** [comment]
Ik heb één inzicht gevonden uit ons onderzoek naar formulieren in 2022.

De radio buttons die we hadden ontworpen waren 44px groot. Gebruikers waardeerden deze grootte, omdat het eenvoudig te selecteren was op touchscreen devices. 

Het is dus belangrijk om rekening te houden met het bieden van voldoende ruimte bij radio groups en/of tussen form field radio options. 

<img width="363" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/580c6cb8-b6d3-43ab-a427-f87970f634f1">

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook (Fieldset)](https://designsystem.amsterdam/?path=/docs/components-forms-radio--docs#examples)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/radio)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=9600-33863&mode=design&t=p24VVZsX9wzPsnkW-4)

---

### Screenshots / Varianten

**Storybook**

Radio group

![Scherm­afbeelding 2024-09-13 om 14 42 42](https://github.com/user-attachments/assets/87706087-0155-4446-a1de-7331614db768)

Radio group with error
![Scherm­afbeelding 2024-09-13 om 14 42 47](https://github.com/user-attachments/assets/241abfa1-b215-4bab-aee5-7f458594532c)


**Figma**
Form field met Radio group
![Scherm­afbeelding 2024-02-27 om 14 43 10](https://github.com/nl-design-system/backlog/assets/545207/ecd70e83-cfaf-4ca7-8d53-45774f24a36a)

Form field met Radio group - validatie
![Scherm­afbeelding 2024-02-27 om 14 43 18](https://github.com/nl-design-system/backlog/assets/545207/ba8ba752-1ea3-4e82-8f9e-f1a6d96b2d52)

---

### Notities
- Documentatie beschikbaar (Storybook)
- Bekijk ook de [GitHub Discussion - Radio button toevoeging van Amsterdam](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10799114).

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/story/css-formfield--radio-button-group)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1681-7054&mode=design&t=f5OPj2AvhbnNdfyi-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=954-1911&mode=design&t=nGE8K2kpeyar960o-4)

---

### Screenshots / Varianten

**Storybook**

Radio Button Group
![Scherm­afbeelding 2024-02-28 om 11 20 04](https://github.com/nl-design-system/backlog/assets/545207/11816b64-03b1-4a02-a31a-c5288140818a)

Radio Button Group With Description
![Scherm­afbeelding 2024-02-28 om 11 20 08](https://github.com/nl-design-system/backlog/assets/545207/4bb3d44f-355a-4719-93c7-8cca359ac555)

**Figma - Huidig**

Radio button group
![Scherm­afbeelding 2024-02-27 om 14 54 22](https://github.com/nl-design-system/backlog/assets/545207/0d276a8e-5d64-4d5e-853c-286bfe2c936a)

**Figma - WIP**

Radio group
![Scherm­afbeelding 2024-02-27 om 14 57 19](https://github.com/nl-design-system/backlog/assets/545207/e11bdbf1-383f-437a-abab-2bf74b3ee32e)

Radio group met description
![Scherm­afbeelding 2024-02-27 om 14 57 58](https://github.com/nl-design-system/backlog/assets/545207/c6e35eaa-e106-4fc5-89c4-ec946e467157)

Mogelijkheid voor description per optie.
![Scherm­afbeelding 2024-02-27 om 14 56 27](https://github.com/nl-design-system/backlog/assets/545207/44ff9ec4-c29d-4d59-9abe-8f84e59e9f50)

Radio group met validatie
![Scherm­afbeelding 2024-02-27 om 14 58 03](https://github.com/nl-design-system/backlog/assets/545207/b105af21-129f-4f8e-a4c3-6456179db3d9)


---

### Notities
- [Zie comment van Rozerin](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8593279) over onderzoek waarbij grotere radio's werden ingezet.
- Huidige Figma component heeft ook een mogelijkheid voor 'Large' variant. Hierbij schaalt het label mee.
- Bekijk ook de [GitHub Discussion - Radio button toevoeging van Den Haag](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10799175).

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-form-groups-radios--radios&viewMode=story)

---

### Storybook

Groups / radios - checked
![Scherm­afbeelding 2024-02-27 om 15 10 29](https://github.com/nl-design-system/backlog/assets/545207/913a1d73-e115-4b51-b923-5b1a944a469c)

Groups / radios - validatie en helptext
![Scherm­afbeelding 2024-02-27 om 15 12 31](https://github.com/nl-design-system/backlog/assets/545207/e62d4cdb-0ef0-421a-a31b-47451a6bc0d9)


Nested combinatie met checkboxes.
![Scherm­afbeelding 2024-02-27 om 15 14 36](https://github.com/nl-design-system/backlog/assets/545207/0a6bb294-cd8f-47d5-beb1-efd42bacaaf2)


---

### Notities
- Optie voor nested weergave.
- Bekijk ook de [GitHub Discussion - Radio button toevoeging van DSO](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10799347).

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/radiobutton)

---

### Screenshots / Varianten

**Documentatie website**

Verticale variant
Radiobuttons worden standaard verticaal weergegeven. Deze vorm zal in de meeste gevallen geschikt zijn.
![Scherm­afbeelding 2024-02-27 om 15 17 22](https://github.com/nl-design-system/backlog/assets/545207/65ff8a3d-606a-48aa-861c-1f178d0a66b5)

Horizontale radio buttons in verticaal formulier
Wanneer je maximaal 2 radio buttons hebt met een korte label, mag je deze om ruimte te besparen in een verticaal formulier naast elkaar zetten.
![Scherm­afbeelding 2024-02-27 om 15 17 27](https://github.com/nl-design-system/backlog/assets/545207/e824ef30-ef6a-43b3-987a-a472e7382ce3)

Horizontale radio buttons in horizontaal formulier
In een horizontaal formulier is het mogelijk meerdere radio buttons naast elkaar te weergeven. Aangeraden wordt om niet meer dan 3 radio buttons te gebruiken. Horizontaal gebruik van formulier elementen is alleen bedoeld voor medewerkersschermen.
![Scherm­afbeelding 2024-02-27 om 15 17 33](https://github.com/nl-design-system/backlog/assets/545207/883b9cf3-0a2f-4413-9bba-42bdd8784428)

Verticale radio buttons in horizontaal formulier
In een horizontaal formulier is het mogelijk meerdere radio buttons onder elkaar weer te geven. Horizontaal gebruik van formulier elementen is alleen bedoeld voor medewerkersschermen.
![Scherm­afbeelding 2024-02-27 om 15 17 42](https://github.com/nl-design-system/backlog/assets/545207/98352d1e-0008-47e6-a4ae-f03ceb0145e7)

Hinttekst
Gebruik hint tekst voor hulp die relevant is voor de meeste gebruikers, zoals hoe hun informatie zal worden gebruikt of waar deze te vinden.
![Scherm­afbeelding 2024-02-27 om 15 17 49](https://github.com/nl-design-system/backlog/assets/545207/09e5a659-5869-4792-b7b8-eee6e2433ecc)

Validatie
![Scherm­afbeelding 2024-02-27 om 15 18 02](https://github.com/nl-design-system/backlog/assets/545207/188dd82c-7e92-409f-af46-762d02448fee)


---


### Notities
- Kader om radio en label, in geheel clickable.
- Veel verschillende mogelijkheden.
- Bekijk ook de [GitHub Discussion - Radio button toevoeging van DUO](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10805227).

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

[Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**

[RadioButton](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-RadioButton-RadioButton/01-molecules-080-Forms-And-Search-RadioButton-RadioButton.html)
![Scherm­afbeelding 2024-02-28 om 11 53 16](https://github.com/nl-design-system/backlog/assets/545207/22a7a4db-5f67-45e9-9d40-a6e9af501d78)

[RadioButton Description](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-RadioButton-RadioButton-Description/01-molecules-080-Forms-And-Search-RadioButton-RadioButton-Description.html)
![Scherm­afbeelding 2024-02-28 om 11 53 20](https://github.com/nl-design-system/backlog/assets/545207/89e22a69-3f76-4f8a-9f02-dc29ed5ac6e5)

[RadioButton Error](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-RadioButton-RadioButton-Error/01-molecules-080-Forms-And-Search-RadioButton-RadioButton-Error.html)
![Scherm­afbeelding 2024-02-28 om 11 53 25](https://github.com/nl-design-system/backlog/assets/545207/e0157b09-188f-44b4-b9cb-9275d087ebff)

---

### Notities
- Form field radio option is ook voorbij het zichtbare label nog 'clickable'.
- Bekijk ook de [GitHub Discussion - Radio button toevoeging van OpenGemeenten](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10805446).

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/apache-wicket-radio-group--docs)
- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=521-28182&mode=design&t=B6BdewKoEAslWiOU-4)

---

### Screenshots / Varianten

**Figma**

Groep - verticaal inclusief validatie
![Scherm­afbeelding 2024-02-27 om 16 49 27](https://github.com/nl-design-system/backlog/assets/545207/afd3c3a2-2c0d-47fe-a384-18004f9af91f)

Groep - horizontaal inclusief validatie
![Scherm­afbeelding 2024-02-27 om 16 49 31](https://github.com/nl-design-system/backlog/assets/545207/171d1276-6cbf-476c-80ae-274d66c84b1e)

---

### Notities
- Onder de noemer: Toggles
- Bekijk ook de [GitHub Discussion - Radio button toevoeging van Rotterdam](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10805460).

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-form-radio-button-group--documentatie)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4234-19768&mode=design&t=hgtnEbtPxQNlUxRM-4)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-radio-button-group)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-02-27 om 17 00 39](https://github.com/nl-design-system/backlog/assets/545207/ab535193-de51-480e-808b-23ee56c1545f)

Invalid
![Scherm­afbeelding 2024-02-28 om 12 02 16](https://github.com/nl-design-system/backlog/assets/545207/a12affbc-0092-483d-8e20-975f32151a00)


**Figma**

Radio group
![Scherm­afbeelding 2024-02-28 om 12 03 11](https://github.com/nl-design-system/backlog/assets/545207/201db51d-253a-4ae0-ac43-fd6811613084)


---

### Notities

Bekijk ook de [GitHub Discussion - Radio button toevoeging van RVO](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10806910).

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css-form-field-radio-group--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/packages/components-react/radio-group-react)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=954-1911&mode=design&t=s1qsA5yiJDvgBH4h-4)

---

### Screenshots / Varianten

**Storybook**

with only a label
![Scherm­afbeelding 2024-02-27 om 17 13 22](https://github.com/nl-design-system/backlog/assets/545207/f39d5c1a-aa41-4f69-a4ee-b8eb87ac7acc)

with heading as label
![Scherm­afbeelding 2024-02-27 om 17 13 27](https://github.com/nl-design-system/backlog/assets/545207/734b7e39-6429-411a-a44c-efa5e8f72a03)

with an additional description
![Scherm­afbeelding 2024-02-27 om 17 13 32](https://github.com/nl-design-system/backlog/assets/545207/45c9ab05-2ed0-4aa5-8a5b-1aa4b7d89dd2)

with a description for an invalid value
![Scherm­afbeelding 2024-02-27 om 17 13 37](https://github.com/nl-design-system/backlog/assets/545207/d5082fc8-b5fc-43ea-906b-1c7cf433b77e)


**Figma**

Radio group
![Scherm­afbeelding 2024-02-27 om 17 16 53](https://github.com/nl-design-system/backlog/assets/545207/8b03a7ae-ef38-4c53-a651-fce5997b5d29)

Radio group met description
![Scherm­afbeelding 2024-02-27 om 17 16 01](https://github.com/nl-design-system/backlog/assets/545207/d05a126b-f684-4fbf-9905-421dbeda0d0a)

Radio group met validatie
![Scherm­afbeelding 2024-02-27 om 17 16 07](https://github.com/nl-design-system/backlog/assets/545207/d4bceb70-af84-4715-8162-414df23bc064)

Mogelijkheid voor description per optie.
![Scherm­afbeelding 2024-02-28 om 12 14 11](https://github.com/nl-design-system/backlog/assets/545207/59eedca2-34d9-42db-8517-099963412220)


---

### Notities
- Figma component heeft eenzelfde description opzet als bij [checkbox group](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8617136).
- Figma component heeft eenzelfde validatie opzet als bij [checkbox group](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8617136).
- Bekijk ook de [GitHub Discussion - Radio button toevoeging van Utrecht](https://github.com/orgs/nl-design-system/discussions/288#discussioncomment-10807078).

**Comment by jeffreylauwers** [comment]
## ✨ **Dit component is nu Help Wanted** ✨

Help je mee hem door de Community stappen te krijgen?
[Radio group op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=radio+group)
